Mage/src/main/java/mage/abilities/keyword/CollectEvidenceAbility.java

```diff
@@ -0,0 +1,100 @@
+package mage.abilities.keyword;
+
+import mage.abilities.Ability;
+import mage.abilities.SpellAbility;
+import mage.abilities.StaticAbility;
+import mage.abilities.condition.common.CollectedEvidenceCondition;
+import mage.abilities.costs.*;
+import mage.abilities.costs.common.CollectEvidenceCost;
+import mage.abilities.hint.ConditionTrueHint;
+import mage.abilities.hint.Hint;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+
+/**
+ * @author TheElk801
+ */
+public class CollectEvidenceAbility extends StaticAbility implements OptionalAdditionalSourceCosts {
+
+    private static final String promptString = "Collect evidence";
+    private static final String keywordText = "As an additional cost to cast this spell, you may collect evidence ";
+    private static final String reminderText = "Exile cards with total mana value $$$ or greater from your graveyard";
+    private final String rule;
+
+    public static final String COLLECT_EVIDENCE_ACTIVATION_VALUE_KEY = "collectEvidenceActivation";
+
+    protected OptionalAdditionalCost additionalCost;
+
+    private static final Hint hint = new ConditionTrueHint(CollectedEvidenceCondition.instance, "evidence was collected");
+
+    public static OptionalAdditionalCost makeCost(int amount) {
+        OptionalAdditionalCost cost = new OptionalAdditionalCostImpl(
+                keywordText + amount,
+                reminderText.replace("$$$", "" + amount),
+                new CollectEvidenceCost(amount)
+        );
+        cost.setRepeatable(false);
+        return cost;
+    }
+
+    public CollectEvidenceAbility(int amount) {
+        super(Zone.STACK, null);
+        this.additionalCost = makeCost(amount);
+        this.rule = additionalCost.getName() + ' ' + additionalCost.getReminderText();
+        this.setRuleAtTheTop(true);
+        this.addHint(hint);
+    }
+
+    private CollectEvidenceAbility(final CollectEvidenceAbility ability) {
+        super(ability);
+        this.rule = ability.rule;
+        this.additionalCost = ability.additionalCost.copy();
+    }
+
+    @Override
+    public CollectEvidenceAbility copy() {
+        return new CollectEvidenceAbility(this);
+    }
+
+    public void resetBargain() {
+        if (additionalCost != null) {
+            additionalCost.reset();
+        }
+    }
+
+    @Override
+    public void addOptionalAdditionalCosts(Ability ability, Game game) {
+        if (!(ability instanceof SpellAbility)) {
+            return;
+        }
+
+        Player player = game.getPlayer(ability.getControllerId());
+        if (player == null) {
+            return;
+        }
+
+        this.resetBargain();
```

Comment:
this should be renamed

Reply:
missing a rename is possibly the most embarrassing thing that can happen